### DISCUSSION TITLE
How can I test a payment gateway plugin without the frontend?

### DISCUSSION THREAD

**Discussion by Abdelmasieh01** [saleor/saleor#15382] [Q&A]
I've made a plugin for a payment gateway called "MyFatoorah". The plugin is available in the plugins api.
I'm working on an edited saleor version and the storefront doesn't seem to work properly with it and I don't have access to the frontend, how can I test this plugin and make a payment without the frontend?

**Comment by timuric** [comment]
> I'm working on an edited saleor version

We highly discourage to fork Saleor, as maintaining a fork of such a large code base is very labor intensive, it also makes it impossible for the community to provide support or be able to upgrade

**Reply by Abdelmasieh01** [reply]
I know that. The client just sent me their forked repo and asked me to add a plugin, the frontend is not currently available to use, so I'm trying to find how to create a checkout through mutations with the API, can you provide the 'normal' saleor steps to creating a payment with the API? The docs haven't been of much help.

**Reply by timuric** [reply]
We recommend building payment app instead of plugin API which we will deprecate in the future https://docs.saleor.io/docs/3.x/developer/payments

**Comment by korycins** [comment]
You can check  doc chapters for [AdyenPlugin](https://docs.saleor.io/docs/3.x/developer/app-store/legacy-plugins/adyen#adding-a-payment-form), [StripePlugin](https://docs.saleor.io/docs/3.x/developer/app-store/legacy-plugins/stripe#creating-payment-in-saleor). It covers the whole API flow, to process the payment with payment-plugin. There is no general way of testing the payment-plugin implementation without storefront. Each payment provider has a different way of handling the process, so you need to figure out your own way of testing it (for example, compare the process for AdyenPlugin and StripePlugin. They're different). You could build a dummy storefront to process the payment, you can modify the sample payment provider storefront to communicate with Saleor, payment provider could also have some docs about testing the backend implementation without storefront.

**Reply by Abdelmasieh01** [reply]
Thank you for replying to this, I already found ways to make a checkout and execute a payment flow with it.
